datanode/networkhistory/README.md

```diff
@@ -0,0 +1,100 @@
+
+
+
+## Why we have it
+
+Gigabytes of event data per day are produced by the Vega core and stored in the Datanode.  It is not feasible for a new Datanode joining the network to replay all blocks from the first block in order to recreate the history and get itself into a state to be able to consume new blocks - such as ensuring it has all parties and markets - it would take far too long.  Network History allows the data-node to reach out to peer nodes to fetch the most recent history,  as well as older history if desired, such that it can quickly get itself up to the latest block height of the network and start to consume events for the latest block from the Vega core.  In additiona decentralization is at the heart of the Vega platform, and thus the mechanism to store and retrieve network history must itself be decentralized.
+
+## How to use it
+
+By default, a new data node will connect to its peers, retrieve the latest segment of history, and load this into the Datanode such that it is in a state to consume new blocks immediately, this is controlled by the configuration parameter `AutoInitialiseFromNetworkHistory` .   By default Datanode is setup to fetch just the latest History Segment, if more history is required that can specified when starting the Datanode for the first time using the `--networkhistory.initialise.block-count` flag, for example to start your new node and have it fetch and load the last days data, assuming a time between blocks of 1s, you would do this:
+
+`data-node node --networkhistory.initialise.block-count=86400`
+
+However, depending on how much history you would like to load, fetching it all synchronously at startup may not be the best approach as the networks block height will continue to increase whilst the history is being fetched, thus we need a way to start the node and have it process and new blocks whilst fetching history asynchronously. How to do this is described in the next section.
+
+If the node already contains history and `AutoInitialiseFromNetworkHistory = true` it will attempt to fetch and load all history between the Datanode's current block height and the network's latest block height and load this into the node before processing new blocks.
+
+### Getting History Asynchronously
+
+After starting the data-node, run:
+
+`data-node network-history show`
+
+This will display the History Segments stored in the node, e.g:
+
+`from_height:2001 to_height:3000 chain_id:"test-chain-clbOFm" history_segment_id:"QmUggHRGorV264FxsbnQUGQpGCTdjd5xytFXUyEnQN1dy1" previous_history_segment_id:"QmaBRouyXMtprM3RyYy6Gbr4QcMmf8j5hnMu3T87wsapyY"`
+
+Select the id of the oldest segment (lowest `from_height`) and then run:
+
+`data-node network-history fetch QmUggHRGorV264FxsbnQUGQpGCTdjd5xytFXUyEnQN1dy1 1000`
+
+Replacing 1000 in the above example with however many blocks of history you would like to fetch and the segment id with the id of the oldest segment in your node.  Once this completes, stop the data-node and run:
```

Comment:
```suggestion
Replace `1000` in the above example with however many blocks of history you would like to fetch and the segment ID with the ID of the oldest segment in your node.  Once this completes, stop the data node and run:
```

@ettec does blocks of history = the number of blocks for which you want history, or is it a different concept, a block of history?

Reply:
its the number of blocks - same thing